packages/ast-spec/tests/fixtures.test.ts

```diff
@@ -0,0 +1,249 @@
+import fs from 'fs';
+import glob from 'glob';
+import makeDir from 'make-dir';
+import path from 'path';
+import { parseBabel } from './util/parsers/babel';
+import { Fixture } from './util/parsers/parser-types';
+import { parseTSESTree } from './util/parsers/typescript-estree';
+import { snapshotDiff, diffHasChanges } from './util/snapshot-diff';
+
+const PACKAGE_ROOT = path.resolve(__dirname, '..');
+const SRC_DIR = path.resolve(PACKAGE_ROOT, 'src');
+
+// Assign a segment set to this variable to limit the test to only this segment
+// This is super helpful if you need to debug why a specific fixture isn't producing the correct output
+// eg. ['declaration', 'ClassDeclaration', 'abstract'] will only test /declaration/ClassDeclaration/fixtures/abstract/fixture.ts
+// prettier-ignore
+const ONLY = [].join(path.sep);
+
+const fixturesWithASTDifferences = new Set<string>();
+const fixturesWithTokenDifferences = new Set<string>();
+enum ErrorLabel {
+  TSESTree = "TSESTree errored but Babel didn't",
+  Babel = "Babel errored but TSESTree didn't",
+  Both = 'Both errored',
+  None = 'No errors',
+}
+const fixturesWithErrorDifferences = {
+  [ErrorLabel.TSESTree]: new Set<string>(),
+  [ErrorLabel.Babel]: new Set<string>(),
+} as const;
+
+const validFixtures = glob.sync(`${SRC_DIR}/**/fixtures/*/*.{ts,tsx}`);
+const errorFixtures = glob.sync(`${SRC_DIR}/**/fixtures/_error_/*/*.{ts,tsx}`);
+
+const fixtures: readonly Fixture[] = [...validFixtures, ...errorFixtures].map(
+  absolute => {
+    const relativeToSrc = path.relative(SRC_DIR, absolute);
+    const { dir, ext } = path.parse(relativeToSrc);
+    const segments = dir.split(path.sep).filter(s => s !== 'fixtures');
+    const name = segments.pop()!;
+    const snapshotPath = path.join(SRC_DIR, dir, 'snapshots');
+    return {
+      absolute,
+      name,
+      ext,
+      isError: absolute.includes('/_error_/'),
+      isJSX: ext.endsWith('x'),
+      relative: path.relative(SRC_DIR, absolute),
+      segments,
+      snapshotPath,
+      snapshotFiles: {
+        success: {
+          tsestree: {
+            ast: path.join(snapshotPath, '1-TSESTree-AST.shot'),
+            tokens: path.join(snapshotPath, '2-TSESTree-Tokens.shot'),
+          },
+          babel: {
+            ast: path.join(snapshotPath, '3-Babel-AST.shot'),
+            tokens: path.join(snapshotPath, '4-Babel-Tokens.shot'),
+          },
+          alignment: {
+            ast: path.join(snapshotPath, '5-AST-Alignment-AST.shot'),
+            tokens: path.join(snapshotPath, '6-AST-Alignment-Tokens.shot'),
+          },
+        },
+        error: {
+          tsestree: path.join(snapshotPath, '1-TSESTree-Error.shot'),
+          babel: path.join(snapshotPath, '2-Babel-Error.shot'),
+          alignment: path.join(snapshotPath, '3-Alignment-Error.shot'),
+        },
+      },
+    };
+  },
+);
+
+function isError(e: unknown): e is { code: unknown } {
```

Comment:
Nit: I would have thought `isError` is checking for `e is Error`. Maybe...

```suggestion
function hasErrorCode(e: unknown): e is { code: unknown } {
```